### ISSUE TITLE
The dword_t vs pointer confusion

### ISSUE THREAD

**Issue by defdefred** [VendelinSlezak/BleskOS#79]
You known what I mean :-)

**Comment by VendelinSlezak** [comment]
Yeah :) I am planning after developing Document editor into usable version to take look at all code to find possible errors, optimalize it, fix this dword_t thing, and also use more structs instead of pointers. I will also take look at possible optimizations by GCC.
